core/src/gl/gl_fwd.h

```diff
@@ -0,0 +1,334 @@
+#pragma once
+
+#ifdef PLATFORM_ANDROID
```

Comment:
The android build is picky about providing its own typedef for GLsizeiptr (due to building with  -pedantic I guess)


OK the last two commit could need a bit more thought - We can postpone them from this PR, though I really like to not have to link GL + GLFW to build the tests..